build-tools/build-indexes

```diff
@@ -913,6 +913,39 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 				overrideAbilityDesc[id] = (pastEntry.shortDesc || pastEntry.desc);
 			}
 		}
+
+		const overrideTypeChart = {};
+		BattleTeambuilderTable[gen].overrideTypeChart = overrideTypeChart;
+		const removeType = {};
+		BattleTeambuilderTable[gen].removeType = removeType;
+		for (const id in nextGenData.TypeChart) {
+			const nextEntry = nextGenData.TypeChart[id];
+			const pastEntry = genData.TypeChart[id];
+			if (!pastEntry) {
+				removeType[id] = true;
+				continue;
+			}
+			if (Object.keys(pastEntry).length !== Object.keys(nextEntry).length) {
+				overrideTypeChart[id] = pastEntry;
+				continue;
+			}
+			let diff = false;
+			for (const key in pastEntry) {
+				if ((typeof pastEntry[key] === 'string' && pastEntry[key] !== nextEntry[key]) ||
+					Object.keys(pastEntry[key]).length !== Object.keys(nextEntry[key]).length) {
+					diff = true;
+					break;
+				}
+				for (const k in pastEntry[key]) {
+					if (nextEntry[key][k] !== pastEntry[key][k]) {
+						diff = true;
+						break;
+					}
+				}
+				if (diff) break;
+			}
+			if (diff) overrideTypeChart[id] = pastEntry;
```

Comment:
I guess this is a deep compare? I agree that it sucks JS doesn't have this built-in, but honestly I would use a hack for readability:

```suggestion
			if (JSON.stringify(nextEntry) !== JSON.stringify(pastEntry)) {
				overrideTypeChart[id] = pastEntry;
			}
```

Reply:
(This is in a build script and will be executed 8 times ever, so I'm prioritizing readability over perf here.)

Reply:
I agree perf isnt important here, and I wanted to do JSON.stringify myself, only it breaks if the keys are in different orders.

Reply:
Are the keys commonly in different orders but otherwise identical?

Reply:
That sounds like a bug in the data files rather than here.

Reply:
I'm not changing this code to introduce subtle bugs if someone lists `spd: 30, spe: 30` vs. `spe: 30, spd: 30`. I'm unable to package the client anyway so I'm happy to drop this change because I don't depend on its correctness anymore, but I don't want my name showing up in `git blame` when theres some stupid and easily preventable bug because you depended on key ordering.

Reply:
TypeChart doesn't contain stats...

I also think you're going backwards? My change preserves key ordering, your change goes out of your way to normalize it.

Reply:
Like, the only reason to prefer your version over mine is if you prioritize a smaller bundle size for `BattleTeambuilderTable` over preserving key ordering, but that's not what you're arguing for at all.

And the smaller bundle size only comes up if someone messes up key ordering in the data files because they are not currently bugged and never intentionally be.

Reply:
This conversation will come up in `git blame`, anyway, but if you _really_ care, I can merge and commit this separately.

Reply:
Youre right, but I dont care to fix this.